solutions/cses-1640.mdx

```diff
@@ -141,3 +123,127 @@ print("IMPOSSIBLE")
 </PySection>
 
 </LanguageSection>
+
+## Solution 2 - Two Pointers
+
+By keeping two pointers, one at each end of the list, we can greedily move the left and right pointers if the sum of the two elements are less than or greater than the target sum, finding the two indices in $\mathcal{O}(N)$ time.
+<LanguageSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class TwoSum {
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+		long N, X;
+		N = io.nextInt();
+		X = io.nextInt();
+
+		List<Pair> a = new ArrayList<>();
+						  
+		// append the element and its index
+		for (int i = 0; i < N; i++) {
+			long x = io.nextInt();
```

Comment:
I think you can just make N X and x of type int since they're not going to exceed the memory limit

Reply:
so just like `long N; int X;`? sorry i don't use java 😰

Reply:
I think N can be an int too right? I haven't read the question though to I'm not 100% sure

Reply:
Ok yeah it has to be an int

Reply:
So you can just do:

int N = io.nextInt();
int X = io.nextInt();

And you don't need the line where you declare N and X.

(Sorry for the lack of formatting I'm on the car rn)

Reply:
nice. thanks for your help!